### PR TITLE
Removing some Guava usages

### PR DESCRIPTION
**Pull request by basil**
See [the mailing list thread](https://groups.google.com/g/jenkinsci-dev/c/aYUJ4VuOuVc). Removing the following Guava usages:

- `com/google/common/collect/ImmutableMap`
- `com/google/common/base/Joiner#join(Ljava/lang/Iterable;)Ljava/lang/String;`
- `com/google/common/base/Joiner#join([Ljava/lang/Object;)Ljava/lang/String;`

### CODE REVIEW

File: src/main/java/hudson/plugins/timestamper/action/TimestampsActionOutput.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ static BufferedReader open(
       for (Function<Timestamp, String> format : query.timestampFormats) {
         parts.add(format.apply(currentTimestamp));
       }
-      String result = Joiner.on(' ').join(parts) + "\n";
+      String result = Util.join(parts, " ") + "\n";
```
Comment (jtnord) [inline src/main/java/hudson/plugins/timestamper/action/TimestampsActionOutput.java:98]: is this not just
```suggestion
      String result = String.join(" ", parts) + "\n";
```

I would have thought that `Util.join` was Deprecated (or should be soon)?

File: src/main/java/hudson/plugins/timestamper/io/TimeShiftsReader.java [hunk 2]
```diff
@@ -72,7 +71,7 @@
    */
   Optional<Long> getTime(long timestampEntry) throws IOException {
     if (timeShifts == null) {
-      timeShifts = ImmutableMap.copyOf(readTimeShifts());
+      timeShifts = Collections.unmodifiableMap(readTimeShifts());
```
Comment (jtnord) [inline src/main/java/hudson/plugins/timestamper/io/TimeShiftsReader.java:74]: this is not a direct replacement.  the returned map can change (even though it is unmodifiable) as it reasds through to the underlying map.

based on the actual implementation of `readTimeShifts` it is ok here though.

File: src/main/java/hudson/plugins/timestamper/io/DumpTimestamps.java [hunk 3]
```diff
@@ -50,7 +51,7 @@ public static void main(String... args) throws IOException {
     if (args.length == 0) {
       throw new IllegalArgumentException("no command-line arguments");
     }
-    File timestamperDir = new File(Joiner.on(' ').join(args));
+    File timestamperDir = new File(Util.join(Arrays.asList(args), " "));
```
Comment (jtnord) [inline src/main/java/hudson/plugins/timestamper/io/DumpTimestamps.java:54]: this looks like somehow someone did not know how to quote arguments :-o

**Review by jtnord — APPROVED** [review]
it does what it says on the tin, but could also be improved

File: src/main/java/hudson/plugins/timestamper/action/TimestampsActionOutput.java (hunk 1 above)
Comment (basil) [reply src/main/java/hudson/plugins/timestamper/action/TimestampsActionOutput.java]: Done

File: src/main/java/hudson/plugins/timestamper/io/TimeShiftsReader.java (hunk 2 above)
Comment (basil) [reply src/main/java/hudson/plugins/timestamper/io/TimeShiftsReader.java:74]: Yes, not an issue based on the actual implementation of `readTimeShifts`.

File: src/main/java/hudson/plugins/timestamper/io/DumpTimestamps.java (hunk 3 above)
Comment (basil) [reply src/main/java/hudson/plugins/timestamper/io/DumpTimestamps.java]: Indeed, but this is a preexisting issue that I am making neither better nor worse in this PR.

**Review by jtnord — APPROVED** [review]

**Pull request merged**